sdks/java/io/clickhouse/src/main/java/org/apache/beam/sdk/io/clickhouse/ClickHouseIO.java

```diff
@@ -0,0 +1,364 @@
+      Properties properties = properties() == null ? new Properties() : properties();
+
+      input.apply(ParDo.of(WriteFn.create(jdbcUrl(), table(), tableSchema, properties)));
+
+      return PDone.in(input.getPipeline());
+    }
+
+    public static <T> Builder<T> builder() {
+      return new AutoValue_ClickHouseIO_Write.Builder<>();
+    }
+
+    /** Builder for {@link Write}. */
+    @AutoValue.Builder
+    public abstract static class Builder<T> {
+
+      public abstract Builder<T> jdbcUrl(String jdbcUrl);
+
+      public abstract Builder<T> table(String table);
+
+      public abstract Builder<T> properties(Properties properties);
+
+      public abstract Write<T> build();
+    }
+  }
+
+  public static ClickHouseProperties properties() {
+    return new ClickHouseProperties();
+  }
+
+  /**
+   * Builder for {@link Properties} for JDBC connection.
+   *
+   * @see <a href="https://clickhouse.yandex/docs/en/single/#settings_1">ClickHouse
+   *     documentation</a>
+   */
+  public static class ClickHouseProperties implements Serializable {
+    private final Properties properties = new Properties();
+
+    /**
+     * Recommendation for what size of block (in number of rows) to load from tables.
+     *
+     * @see <a href="https://clickhouse.yandex/docs/en/single/#max_block_size">ClickHouse
+     *     documentation</a>
+     * @param value number of rows
+     * @return builder
+     */
+    public ClickHouseProperties maxBlockSize(int value) {
+      return set(ClickHouseQueryParam.MAX_BLOCK_SIZE, value);
+    }
+
+    /**
+     * The maximum block size for insertion, if we control the creation of blocks for insertion.
+     *
+     * @see <a href="https://clickhouse.yandex/docs/en/single/#max_insert_block_size">ClickHouse
+     *     documentation</a>
+     * @param value number of rows
+     * @return builder
+     */
+    public ClickHouseProperties maxInsertBlockSize(long value) {
+      return set(ClickHouseQueryParam.MAX_INSERT_BLOCK_SIZE, value);
+    }
+
+    /**
+     * If setting is enabled, insert query into distributed waits until data will be sent to all
+     * nodes in cluster.
+     *
+     * @param value true to enable
+     * @return builder
+     */
+    public ClickHouseProperties insertDistributedSync(boolean value) {
+      return set("insert_distributed_sync", value ? 1 : 0);
+    }
+
+    /**
+     * For INSERT queries in the replicated table, wait writing for the specified number of replicas
+     * and linearize the addition of the data. 0 - disabled.
+     *
+     * @see <a href="https://clickhouse.yandex/docs/en/single/#insert_quorum">ClickHouse
+     *     documentation</a>
+     * @param value number of replicas, 0 for disabling
+     * @return builder
+     */
+    public ClickHouseProperties insertQuorum(long value) {
+      return set(ClickHouseQueryParam.INSERT_QUORUM, value);
+    }
+
+    /**
+     * For INSERT queries in the replicated table, specifies that deduplication of inserting blocks
```

Comment:
This setting is enabled by default, and it used only by ClickHouse, but not the connector. Probably we should completely remove it to avoid confusion.